pex/util.py

```diff
@@ -177,6 +177,9 @@ def cache_distribution(cls, zf, source, target_dir):
           with contextlib.closing(zf.open(name)) as zi:
             with safe_open(os.path.join(target_dir_tmp, target_name), 'wb') as fp:
               shutil.copyfileobj(zi, fp)
+              fileinfo = zf.getinfo(zi.name)
```

Comment:
Maybe extract these 2 lines into a utility function in this file that pex_builder.py can re-use.